python/brainyflow.py

```
@@ -118,95 +132,101 @@
     
-    async def run(self, memory, propagate=False):
+    @overload
+    async def run(self, memory: Union[Memory[G, L], G], propagate: Literal[True]) -> List[Tuple[Action, Memory[G, L]]]: ...
+    
+    @overload
+    async def run(self, memory: Union[Memory[G, L], G], propagate: Literal[False] = False) -> ExecResultT: ...
+    
+    async def run(self, memory: Union[Memory[G, L], G], propagate: bool = False) -> Union[List[Tuple[Action, Memory[G, L]]], ExecResultT]:
         """Run the node's full lifecycle."""
         if self.successors:
             warnings.warn("Node won't run successors. Use Flow!")
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)